articles/google-play-in-app-reviews/index.md

```diff
@@ -0,0 +1,250 @@
+                public void onComplete(Task<Void> task) {
+                    Toast.makeText(getApplicationContext(), "In App Rating complete", Toast.LENGTH_LONG).show();
+                }
+            });
+        }
+        else {
+            Toast.makeText(getApplicationContext(), "In App Rating failed", Toast.LENGTH_LONG).show();
+        }
+    }
+}
+```
+
+### Testing
+To test our application, we will use internal app sharing. To confirm that this works, add a `Button` to trigger the review flow. Remember, we are using a call to action for testing. In production applications, calls to action such as `Button` should be avoided when using this API.
+
+- Add a button in your activity.xml
+
+```xml
+<Button
+    android:id="@+id/button"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginStart="159dp"
+    android:layout_marginEnd="164dp"
+    android:layout_marginBottom="600dp"
+    android:text="SHOW IN APP REVIEW DIALOG"
+    app:layout_constraintBottom_toBottomOf="parent"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent" />
+```
+
+- Declare the button
+
+```java
+Button mButton;
+```
+
+Below `onCreate`,
+
+- Create an instance of the `mButton`
+
+```java
+mButton = findViewById(R.id.button);
+```
+
+- Add `OnClickListener` and call `startReviewFlow()`
+
+```java
+mButton.setOnClickListener(new View.OnClickListener() {
+    @Override
+    public void onClick(View view) {
+        startReviewFlow();
+    }
+});
+```
+
+#### What you need
+- A Google Play Console account.
+- An application already published in the Google Play store.
+
+#### How to test
+1. Generate a signed app bundle/APK. Note the APK signing key and the `applicationId` should be the same as the already published application.
+
+2. Share the generated APK with a tester. To do that, select the published application in the Google console, navigate to [`Internal App Sharing`](https://play.google.com/console/internal-app-sharing), and upload the generated APK there.
+
+2. Copy the upload's shareable link and share it with a tester. In this case, a tester should be using an Android mobile phone.
+
+3. Open the shared link on the phone's browser. You will be prompted to open with the Google Play store. Do so. This will launch a Play store screen.
+
+4. Download the app and wait for the installation to complete.
+
+5. Launch the application and click `SHOW IN APP REVIEW DIALOG` to start the review flow, as shown in the image below.
+
+![image](/engineering-education/google-play-in-app-reviews/in-app-reviews.png)
+
+**Note:**
+- Since we a carrying out a test, the submit button will be inactive.
+- If you have previously reviewed the application, delete your comment in the Google Play store. Otherwise, the review sheet won't pop up.
+
+I hope this guide is a helpful source on how to ask users for reviews as well as implement in-app reviews within your application.
```

Comment:
Please try and add a more detailed conclusion to give a summary of what people should have learned from the article and why. Some kind of closing remarks. If you could host the code on GitHub it would be great to help me with the tech review too.